mkt/webapps/models.py

```diff
@@ -1803,57 +1803,20 @@ def get_cached_manifest(self, force=False):
 
         If the addon is not a packaged app, this will not cache anything.
 
+        Ensure that the calling method checks various permissions if needed.
+        E.g. see mkt/detail/views.py. This is also called as a task after
+        reviewer approval so we can't perform some checks here.
         """
         if not self.is_packaged:
             return
 
-        key = 'webapp:{0}:manifest'.format(self.pk)
```

Comment:
I moved the contents of this method to `utils.py` to share it with langpacks. There was a big ugly bug (that is now fixed and tested) : caching never worked, because `key` here was overwritten in the loop on the manifest below, right before saving.